pywinauto/unittests/test_uiawrapper.py

```diff
@@ -1251,6 +1252,72 @@ def tearDown(self):
             """Close the application after tests"""
             self.app.kill()
 
+
+    class ComboBoxTestsWinForms(unittest.TestCase):
+
+        """Unit tests for the ComboBoxWrapper class with WinForms app"""
+
+        def setUp(self):
+            """Set some data and ensure the application is in the state we want"""
+            _set_timings()
+
+            # start the application
+            app = Application(backend='uia')
+            self.app = app.start(winfoms_app_grid)
+            self.dlg = dlg = app.Form1
+
+            self.combo_editable = dlg.child_window(auto_id="comboRowType", control_type="ComboBox").wrapper_object()
+            self.combo_fixed = dlg.child_window(auto_id="comboBoxReadOnly", control_type="ComboBox").wrapper_object()
+            self.combo_simple = dlg.child_window(auto_id="comboBoxSimple", control_type="ComboBox").wrapper_object()
+
+        def tearDown(self):
+            """Close the application after tests"""
+            self.app.kill()
+
+        def test_expand_collapse(self):
+            """Test methods .expand() and .collapse() for WinForms combo box"""
+            self.dlg.set_focus()
+            for combo in [self.combo_editable, self.combo_fixed, self.combo_simple]:
+                if combo != self.combo_simple:
+                    self.assertFalse(combo.is_expanded())
+                # test that method allows chaining
+                self.assertEqual(combo.expand(), combo)
+                self.assertTrue(combo.is_expanded())
+
+                # .expand() keeps already expanded state (and still allows chaining)
+                self.assertEqual(combo.expand(), combo)
+                self.assertTrue(combo.is_expanded())
+                
+                # collapse
+                self.assertEqual(combo.collapse(), combo)
+                if combo != self.combo_simple:
+                    self.assertFalse(combo.is_expanded())
+                
+                # collapse already collapsed should keep collapsed state
+                self.assertEqual(combo.collapse(), combo)
+                if combo != self.combo_simple:
+                    self.assertFalse(combo.is_expanded())
+
```

Comment:
Since we loop through several combos in this test, adding current combo description in all asserts could save time when tracing down the failure.

Reply:
Already faced with this. Will eliminate the loop.

Reply:
OK, for one case I will add `msg` param.